analyses/chromosomal-instability/util/chr-break-calculate.R

```diff
@@ -0,0 +1,233 @@
+    stop("No breaks data.frame has been provided. ")
+  }
+
+  # List the columns we need
+  needed_cols <- c(samples_col, chrom_col, start_col, end_col)
+
+  # Get logical vector indicating which are in metadata
+  found_cols <- (needed_cols %in% colnames(break_df))
+
+  # If not all the columns are found, stop
+  if (!all(found_cols)) {
+    stop(cat(
+      "The following column names specified for making the GenomicRanges object: \n",
+      paste(needed_cols[which(!found_cols)], collapse = "\n"),
+      "\n ...were not found in the specified metadata data.frame.",
+      "Check your `_col` arguments."
+    ))
+  }
+
+  if (sample_id[1] != "all") {
+    # Extract samples' info from the data.frame
+    break_df <- break_df %>%
+      dplyr::filter(!!rlang::sym(samples_col) %in% c(sample_id))
+
+    # Stop if the sample doesn't exist in the data.frame
+    if (nrow(break_df) == 0) {
+      stop("No sample data was found for the given `sample_id` argument.")
+    }
+  }
+
+  # Make GRanges for CNV data
+  granges <- GenomicRanges::GRanges(
+    seqnames = dplyr::pull(break_df, !!rlang::sym(chrom_col)),
+    ranges = IRanges::IRanges(
+      start = dplyr::pull(break_df, !!rlang::sym(start_col)),
+      end = dplyr::pull(break_df, !!rlang::sym(end_col))
+    ),
+    mcols = dplyr::pull(break_df, !!rlang::sym(samples_col))
+  )
+
+  return(granges)
+}
+
+break_density <- function(sv_breaks = NULL,
+                          cnv_breaks = NULL,
+                          sample_id = NULL,
+                          window_size = 1e6,
+                          chr_sizes_list = NULL,
+                          samples_col_cnv = "samples",
+                          chrom_col_cnv = "chrom",
+                          start_col_cnv = "start",
+                          end_col_cnv = "end",
+                          samples_col_sv = "samples",
+                          chrom_col_sv = "chrom",
+                          start_col_sv = "start",
+                          end_col_sv = "end") {
+  # For given breaks data.frame(s), calculate the breaks density for a tiled
+  # windows across the genome. Returns the data as a GenomicRanges object for
+  # easy mapping with `ggbio`. Where the density and counts are stored in
+  # @elementMetadata@listData.
+  #
+  # Args:
+  #   sv_breaks: a data.frame with the breaks for the SV data.
+  #   cnv_breaks: a data.frame with the breaks for the SV data.
+  #   sample_id: sample ID for which the data needs to be extracted and made into
+  #              a GenomicRanges object. If "all" is designated, all samples will
+  #              be kept. Multiple samples can be designated as a character vector.
+  #   chr_size_list: a named numeric vector of the sizes (bp) of the chromosomes.
+  #                  names of the chromosomes must match the format of the input
+  #                  break data. e.g. "chr1" or "1".
+  #   window_size: What size windows to calculate break density. Default is 1kb.
```

Comment:
Your default is 1e6 which feels more than 1kb? Something needs to change, either description to mb or default to 1e3.

Reply:
Ah. This is a remnant I missed. Did have it 1kb, then switched it but forgot some of the labels.